### PR TITLE
Migrate RCHTTPRequest

### LINKED ISSUE #582: Migrate RCHTTPRequest

### PR DESCRIPTION
**Pull request by Juanpe**
Resolves #582 

The goal of this PR is to migrate the `RCHTTPRequest` object to `Swift`. I added some // TODO to make it internal when the migration is done.

### CODE REVIEW

File: PurchasesCoreSwift/Networking/HTTPRequest.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            path: path,
+            requestBody: requestBody,
+            headers: headers,
+            completionHandler: completionHandler
+        )
+
+        return copy
+    }
+
+    public override var description: String {
```
Comment (Juanpe) [inline PurchasesCoreSwift/Networking/HTTPRequest.swift:43]: Previous output:
```
<RCHTTPRequest: httpMethod=POST
path=/a_random_path
requestBody={
    requestNumber = 1;
}
headers=(null)
>
```

New output:
```
<HTTPRequest: httpMethod=POST
path=/a_random_path
requestBody=["requestNumber": 1]
headers=(null)
>
```

Comment (taquitos) [reply PurchasesCoreSwift/Networking/HTTPRequest.swift:43]: Yeah, that seems to be what Swift will do. I think this is reasonable.

File: PurchasesCoreSwift/Networking/HTTPRequest.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//
+
+import Foundation
+
+public typealias HTTPClientResponseHandler = (Int, [String: Any]?, NSError?) -> Void
+
+// TODO(post migration): Change back to internal
+@objc(RCHTTPRequest)
+public class HTTPRequest: NSObject, NSCopying {
+    @objc public var httpMethod: String
```
Comment (taquitos) [inline PurchasesCoreSwift/Networking/HTTPRequest.swift:16]: These all can be `let` 🎉

**Review by taquitos — CHANGES_REQUESTED** [review]
One comment then 🚢

File: PurchasesCoreSwift/Networking/HTTPRequest.swift (hunk 2 above)
Comment (Juanpe) [reply PurchasesCoreSwift/Networking/HTTPRequest.swift:16]: Firstly, all these were `let`, but when I checked the previous code, I realized that the properties weren't `readonly`, so I used `var` instead. TBH, I agree with you, let me change them 👍🏼

Comment (taquitos) [reply PurchasesCoreSwift/Networking/HTTPRequest.swift]: Our approach has been to basically fix small things like that where it doesn't impact public API 😄
Public API must stay the same, if we can.

**Review by taquitos — APPROVED** [review]
🎈🐐

**Pull request merged**